optimizely/odp/odp_config.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+from threading import Lock
+
+
+class OdpConfig:
+    """
+    Contains configuration used for ODP integration.
+
+    Args:
+        api_host: The host URL for the ODP audience segments API (optional).
+        api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+        segments_to_check: An array of all ODP segments used in the current datafile
+                            (associated with api_host/api_key).
+    """
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        api_host: Optional[str] = None,
+        segments_to_check: Optional[list[str]] = None
+    ) -> None:
+        self.api_key = api_key
+        self.api_host = api_host
+        self.segments_to_check = segments_to_check or []
+        self.lock = Lock()
+
+    def update(self, api_key: Optional[str], api_host: Optional[str], segments_to_check: list[str]) -> bool:
+        """
+        Override the ODP configuration.
+
+        Args:
+            api_host: The host URL for the ODP audience segments API (optional).
+            api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+            segments_to_check: An array of all ODP segments used in the current datafile
+                               (associated with api_host/api_key).
+
+        Returns:
+            True if the provided values were different than the existing values.
+        """
+        updated = False
+        with self.lock:
+            if self.api_key != api_key or self.api_host != api_host or self.segments_to_check != segments_to_check:
```

Comment:
I prefer to use "not" as in more pythonic....what u think?
Visually ! looks better?

Reply:
I think it reads more like english with `self.api_key != api_key` vs `not self.api_key == api_key`. I think `not` works better with identity comparisons ie  `self.api_key is not api_key` or flipping a bool `not updated`.

Also adding three nots would push us over the line limit 😃

Reply:
I agree. Python's "not" is not positioned intuitively in equality evaluation